woodwork/logical_types.py

```diff
@@ -201,6 +201,15 @@ class Boolean(LogicalType):
 
     primary_dtype = "bool"
 
+    def transform(self, series, null_invalid_values=False):
+        if str(series.dtype) == "object":
+            series = _coerce_boolean(series)
+            if not _is_dask_series(
+                series
+            ):  # Dask cannot handle NA as an ambiguous type for booleans
+                series = series.replace(ww.config.get_option("nan_values"), pd.NA)
+        return super().transform(series)
```

Comment:
We can possibly refactor this to use the pandas method `convert_dtype`

```python
def transform(self, series, null_invalid_values=False):
        if str(series.dtype) == "object" and isinstance(series, pd.Series):
            series = series.convert_dtypes()
        return super().transform(series)
```

Reply:
I'll look into this thanks man!

Reply:
Is this change going in @ParthivNaresh ?